### PR TITLE
Deprecate the project

### PR DESCRIPTION
**Pull request by pitag-ha**
I'm deprecating the project.

We've had the impression that the community is behind it by observing strongly decreased interest in latest compiler support for a while. Now, I've tried to confirm that impression by writing a [discuss post](https://discuss.ocaml.org/t/rfc-deprecating-ocaml-migrate-parsetree-in-favor-of-ppxlib-also-as-a-platform-tool/13240) about deprecating it giving quite some context and asking for potential feedback. Not that many people have read it (< 300 people), but out of the ones that have, nobody has objected or commented :)

Deprecating it, also means that we won't add OCaml 5.1 support or newer anymore.

### CODE REVIEW

File: README.md
Comment (panglesd) [inline README.md]: Nitpick: I think the deprecation warning should be more visible. I don't think `[...]` is a markdown syntax for emphasizing a word. This [post](https://medium.com/maintainer-io/how-to-deprecate-a-repository-on-github-8f0ceb9155e) suggests changing the title, but maybe using bold face and caps lock for `deprecated` is enough?

```suggestion

**DEPRECATED** This package is deprecated. Please, use
```

File: README.md
Comment (panglesd) [inline README.md:4]: You gave some context in the PR, maybe it's worth linking to it?

```suggestion
[ppxlib](https://github.com/ocaml-ppx/ppxlib) instead. See [#126](https://github.com/ocaml-ppx/ocaml-migrate-parsetree/pull/126).
```

**Review by panglesd — APPROVED** [review]
I think you have been cautious enough to make the deprecation move!

[This post](https://medium.com/maintainer-io/how-to-deprecate-a-repository-on-github-8f0ceb9155e) also suggests updating the repository description.

File: README.md
Comment (pitag-ha) [reply README.md:4]: Yes, good idea! I've just opened a PR for an ocaml.org Changelog entry about the deprecation: https://github.com/ocaml/ocaml.org/pull/1712 . I'll link to the Chagelog entry here once that's merged. That contains more detail than the PR description.

**Comment by pitag-ha** [comment]
> [This post](https://medium.com/maintainer-io/how-to-deprecate-a-repository-on-github-8f0ceb9155e) also suggests updating the repository description.

Very good point as well, thanks! I'll add the same note to the repo description as to the README once the ocaml.org Changelog entry is merged.

**Comment by pitag-ha** [comment]
@panglesd, thanks for your suggestions! I've done them all, I think. Does it look good to you?

**Review by panglesd — APPROVED** [review]
Looks all good to me!

**Pull request merged**
